core/src/com/agateau/pixelwheels/gameinput/DigitalSteering.java

```diff
@@ -19,18 +19,27 @@
 package com.agateau.pixelwheels.gameinput;
 
 import com.agateau.pixelwheels.GamePlay;
-import com.agateau.pixelwheels.racescreen.HudButton;
 import com.badlogic.gdx.math.MathUtils;
 
-class TouchInputUtils {
-    static float applyDirectionInput(HudButton leftButton, HudButton rightButton, float direction) {
-        if (leftButton.isPressed()) {
-            direction += GamePlay.instance.steeringStep;
-        } else if (rightButton.isPressed()) {
-            direction -= GamePlay.instance.steeringStep;
-        } else {
-            direction = 0;
-        }
-        return MathUtils.clamp(direction, -1, 1);
+public class DigitalSteering {
+
+  private float rawDirection = 0;
+
+  public float steer(boolean left, boolean right) {
+    if (left && right) {
+      // full steer in original direction,
+      // assuming player intended to reverse at that direction
+      rawDirection = Math.signum(rawDirection);
```

Comment:
I don't understand this part. I think direction should be unchanged if both left & right are pressed.

Reply:
I've added more reasoning as more elaborated comments there, please review and advise further.